File: src/common/py/core/messaging/message_bus_protocol.py
from typing import Protocol

from .message import Message
from .meta import MessageMetaInformationType


class MessageBusProtocol(Protocol):
    # pylint: disable=missing-function-docstring,too-few-public-methods
    """
    Defines the general interface for the ``MessageBus``.
    """
    def dispatch(self, msg: Message, msg_meta: MessageMetaInformationType) -> None: ...

File: src/common/web/core/messaging/MessageBusProtocol.ts
import { Message } from "./Message";
import { MessageMetaInformation } from "./meta/MessageMetaInformation";

/**
 * Defines the general interface for the ``MessageBus``.
 */
export interface MessageBusProtocol {
    dispatch(msg: Message, msgMeta: MessageMetaInformation): void;
}
